packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts

```diff
@@ -109,26 +108,13 @@ export class AmplifyAuthTransform extends AmplifyCategoryTransform {
     const overrideDir = path.join(backendDir, this._category, this.resourceName);
     const isBuild = await buildOverrideDir(backendDir, overrideDir);
     if (isBuild) {
-      const overrideCode: string = await fs.readFile(path.join(overrideDir, 'build', 'override.js'), 'utf-8').catch(() => {
-        formatter.list(['No override File Found', `To override ${this.resourceName} run amplify override auth`]);
-        return '';
-      });
-
-      const sandboxNode = new vm.NodeVM({
-        console: 'inherit',
-        timeout: 5000,
-        sandbox: {},
-        require: {
-          context: 'sandbox',
-          builtin: ['path'],
-          external: true,
-        },
-      });
+      const overrideJSFilePath = path.join(overrideDir, 'build', 'override.js');
       const projectInfo = getProjectInfo();
       try {
-        await sandboxNode
-          .run(overrideCode, path.join(overrideDir, 'build', 'override.js'))
-          .override(this._authTemplateObj as AmplifyAuthCognitoStack & AmplifyStackTemplate, projectInfo);
+        const overrideImport = await import(overrideJSFilePath);
+        if (overrideImport && overrideImport?.override && typeof overrideImport?.override === 'function') {
+          overrideImport.override(this._authTemplateObj, projectInfo);
+        }
```

Comment:
now that all these overrides look very similar consider extracting common code somewhere here https://github.com/aws-amplify/amplify-cli/tree/dev/packages/amplify-cli-core/src/overrides-manager . e.g. create `override-runner.ts` and extract code in such a way that what remains here is `await runOverride(overrideDir, this._authTemplateObj)`.
Project info can be looked up in `core` I think, if not add one more parameter.